src/CTA.Rules.Actions/ActionHelpers/ConfigMigrate.cs

```diff
@@ -67,76 +71,105 @@ private XDocument LoadWebConfig(string projectDir)
+                return defaultContent;
+            }
+            else
             {
-                config.Remove(config.Length - 1, 1);
-                config.Insert(0, "\": { ");
-                config.Insert(0, Constants.ConnectionStrings);
-                config.Insert(0, "\"");
-                config.Append("},");
+                return null;
             }
+        }
 
-            webConfig.Descendants()?.First(d => d.Name?.ToString().ToLower() == Constants.appSettings)?.Descendants()?.ToList().ForEach(appSetting =>
+        /// <summary>
+        /// Gets connection strings from web.config file
+        /// </summary>
+        /// <param name="webConfig"></param>
+        /// <returns></returns>
+        private Dictionary<string, string> GetConnectionStrings(Configuration webConfig)
+        {
+            Dictionary<string, string> connectionStringObjects = new Dictionary<string, string>();
+            var connectionStrings = webConfig.ConnectionStrings?.ConnectionStrings;
+
+            if (connectionStrings != null)
             {
-                try
+                foreach (ConnectionStringSettings connectionString in connectionStrings)
                 {
-                    var key = appSetting.Attributes()?.First(a => a.Name?.ToString().ToLower() == Constants.key).Value;
-                    var value = appSetting.Attributes()?.First(a => a.Name?.ToString().ToLower() == Constants.value).Value;
-                    if (!Constants.appSettingsExclusions.Contains(key))
-                    {
-                        config.Append("\"").Append(key).Append("\"").Append(":").Append("\"").Append(value).Append("\"").Append(",");
-                    }
+                    connectionStringObjects.Add(connectionString.Name, connectionString.ConnectionString);
+                    _hasData = true;
                 }
-                catch (Exception ex)
-                {
-                    LogHelper.LogError(ex, "Error while parsing appsettings");
-                }
-            });
+            }
 
-            return config.ToString();
+            return connectionStringObjects;
         }
 
+        /// <summary>
+        /// Gets app settings from web.config file
+        /// </summary>
+        /// <param name="webConfig"></param>
+        /// <returns></returns>
+        private Dictionary<string, string> GetAppSettingObjects(Configuration webConfig)
+        {
+            Dictionary<string, string> appSettingsObjects = new Dictionary<string, string>();
+            var appSettings = webConfig.AppSettings?.Settings;
 
+            if (appSettings != null)
+            {
+                foreach (KeyValueConfigurationElement appSetting in appSettings)
+                {
+                    if (!Constants.appSettingsExclusions.Contains(appSetting.Key))
+                    {
+                        appSettingsObjects.Add(appSetting.Key, appSetting.Value);
+                    }
+                    _hasData = true;
```

Comment:
If taking other suggestion about assigning `_hasData` in `ProcessWebConfig()` method, we can remove this. If not, ignore this comment.